src/middlewared/middlewared/plugins/pool.py

```diff
@@ -3775,6 +3777,17 @@ def transform(dataset):
         if verrors:
             raise verrors
 
+        if not data.get('managedby') and app:
+            uri = None
+            if app.rest and app.host:
+                uri = app.host
+            elif app.websocket and app.request.headers.get('X-Real-Remote-Addr'):
+                uri = app.request.headers.get('X-Real-Remote-Addr')
+            if uri and uri not in [
+                '::1', '127.0.0.1', *[d['address'] for d in await self.middleware.call('interface.ip_in_use')]
+            ]:
+                data['managedby'] = uri
```

Comment:
Reading ticket description, I see that this field will be used to determine if this dataset has been created by TrueCommand. I can't understand how will we distinguish TrueCommand from Web UI in this case, will we match `managedby` with IPv4|IPv6 regex? Seems awkward.

Reply:
Not just TrueCommand, any API users, so they can identify themselves via name. IP should be a fallback in case no string is specified.